pkg/policy/opa/engine.go

```diff
@@ -249,14 +249,20 @@ func (e *Engine) CompileRegoFiles() error {
 
 // Init initializes the Opa engine
 // Handles loading all rules, filtering, compiling, and preparing for evaluation
-func (e *Engine) Init(policyPath string) error {
+func (e *Engine) Init(policyPath string, scanRules, skipRules []string) error {
 	e.context = context.Background()
 
 	if err := e.LoadRegoFiles(policyPath); err != nil {
 		zap.S().Error("error loading rego files", zap.String("policy path", policyPath), zap.Error(err))
 		return err
 	}
 
+	// before compiling the rego files, filter the rules based on scan and skip rules supplied
+	filterRules(e, policyPath, scanRules, skipRules)
+
+	// update the rule count
+	e.stats.ruleCount = len(e.regoDataMap)
```

Comment:
May be it's a good idea to error out if the `ruleCount` is less than 1.

Reply:
multiple engines are created based on policy path slice, cannot error out.